### PR TITLE
Add API to Add/Remove virtual device

### PR DESCRIPTION
**Pull request by jagadishmurugan**

### CODE REVIEW

**Comment by jagadishmurugan** [comment]
Verified the changes using wssdagent and the following sequence:

-> Create HCS VM on host with 'NVIDIA TESLA T4' GPU
-> Verify - VM is in Running state, has proper IP address, login to the VM and verify GpuDevice is visible in the VM (lspci -v), note down vmBusID (returned by AssignDevice)
-> Stop VM
-> Verify - VM is in Offline state, RemoveDevice succeeds with the vmBusID obtained in Start

Repeat above steps - Make sure the GpuDevice is visible in the VM after Start...

File: vm.go [hunk 1]
```diff
@@ -525,6 +528,74 @@ func (vm *VirtualMachineSpec) UpdateGpuConfiguration(mode GpuAssignmentMode, all
...
+	guid, err := guid.NewV4()
+	if err != nil {
+		return "", err
+	}
+
+	vmBusGUID := guid.String()
+	targetDevice := hcsschema.VirtualPciDevice{
+		Functions: []hcsschema.VirtualPciFunction{
+			{
+				DeviceInstancePath: deviceID,
```
Comment (johnsonshih) [inline vm.go:548]: >deviceID [](http://example.com/codeflow?start=24&length=8)

what happens if the device id already added to the compute system?

File: vm.go [hunk 2]
```diff
@@ -525,6 +528,74 @@ func (vm *VirtualMachineSpec) UpdateGpuConfiguration(mode GpuAssignmentMode, all
...
+	}
+	request := &hcsschema.ModifySettingRequest{
+		ResourcePath: fmt.Sprintf("VirtualMachine/Devices/VirtualPci/%s", vmBusGUID),
+		RequestType:  requesttype.Add,
+		Settings:     targetDevice,
+	}
+
+	// WCOW (when supported) does not require a guest request as part of the
+	// device assignment
+	//if system.os != "windows" {
```
Comment (johnsonshih) [inline vm.go]: remove #Closed

File: vm.go (hunk 1 above)
Comment (jagadishmurugan) [reply vm.go:548]: will need to check that behavior on real hardware... will update once i find out...

**Pull request merged**